src/main/java/com/rultor/agents/IndexesRequests.java

```diff
@@ -39,8 +45,103 @@
  * @version $Id$
  */
 public final class IndexesRequests implements SuperAgent {
+    /**
+     * Xpath expression for retrieving log entries from a talk's XML.
+     */
+    private static final String ARCHIVE_LOG = "//archive/log";
+
+    /**
+     * Name of the log index attribute.
+     */
+    private static final String INDEX = "index";
+
     @Override
+    @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
     public void execute(final Talks talks) throws IOException {
-        throw new UnsupportedOperationException();
+        int maxTalk = this.getMaxTalkIndex(talks);
+        for (final Talk talk : talks.active()) {
+            final List<String> requests = talk.read().xpath("//request");
+            if (requests.isEmpty()) {
+                int index = 0;
+                final List<XML> logs = talk.read().nodes(ARCHIVE_LOG);
+                if (logs.isEmpty()) {
+                    index = maxTalk + 1;
+                } else {
+                    index = this.getMaxLogIndex(logs) + 1;
+                }
+                talk.modify(
+                    new Directives().xpath("//talk").add("request")
+                        .attr(INDEX, Integer.toString(index))
+                        .attr("id", this.createRequestId())
+                        .add("type").set(INDEX)
```

Comment:
The test method `storesIndexIfNone` is not correct. We should NOT do anything if there are no '/talk/request` nodes. The logic is simple, we should set the `/talk/request/@index` to the maximum number + 1. The maximum number we find by searching through all talks and reading their `/archive/log/@index` and `/talk/request/@index` numbers.